public/scripts/helper.js

```diff
@@ -0,0 +1,78 @@
+const cardsContainer = document.querySelector('.card_container');
+const category_id = document.querySelector('.category_id'); 
+const flavor_id = document.querySelector('.flavor_id');
+const createCard = (data) => {
+  cardsContainer.textContent = '';
+  data.forEach((element) => {
+    const card = document.createElement('div');
+    card.classList.add('card');
+
+    const cover = document.createElement('img');
+    cover.classList.add('cover');
+    
+    cover.setAttribute('src',element.img_url);
+
+    const name = document.createElement('h3');
+    name.classList.add('name');
+    name.textContent = element.name;
+
+    const quantity = document.createElement('p');
+    quantity.classList.add('quantity');
+    quantity.textContent ="Available Quantity:" + element.quantity;
+    
+    const category_name = document.createElement('p');
+    category_name.classList.add('category');
+    category_name.textContent = element.category_name;
+
+    const flavor_name = document.createElement('p');
+    flavor_name.classList.add('flavor');
+    flavor_name.textContent = element.flavor_name;
+
+    const delete_button = document.createElement('button');
+    delete_button.classList.add('addCandyButton');
+    delete_button.classList.add('delete');
+    delete_button.setAttribute('type','button');
+    delete_button.textContent = 'delete';
+    delete_button.addEventListener('click',()=>{
+      fetchDelete(`/delete/${element.id}`)
+      // window.location.href = "/";
+    } 
+    
+    );
+    card.append(cover, name, quantity, category_name, flavor_name,delete_button);
+
+    cardsContainer.append(card);
+  });
+};
+
+const createList=(type,data)=>{
+  
+  data.forEach((element)=>{
+    let option = document.createElement('option');
+    option.textContent = element.name;
+    option.value = element.name; 
+    if (type === "category"){
+      category_id.append(option);
+    }else if (type === "flavor"){
+      flavor_id.append(option);
+    }
+    
+  }
+  );
+}; 
+
+const createListWithId=(type,data)=>{
+  
+  data.forEach((element)=>{
+    let option = document.createElement('option');
+    option.textContent = element.name;
+    option.value = element.id; 
+    if (type === "category"){
+      category_id.append(option);
+    }else if (type === "flavor"){
+      flavor_id.append(option);
+    }
+    
+  }
+  );
+}; 
```

Comment:
Both createList() and createListWithId() functions are very similar in structure and could be combined into one function by adding a third argument to indicate whether to use name or id as the option value.

Here's an example of how you can refactor the code to combine createList() and createListWithId() functions into a single function:

```js
const createList = (type, data, useId) => {
  data.forEach((element) => {
    const option = document.createElement('option');
    option.textContent = element.name;
    option.value = useId ? element.id : element.name; 

    if (type === "category") {
      category_id.append(option);
    } else if (type === "flavor") {
      flavor_id.append(option);
    }
  });
};

// Create list with id
createList("category", categoriesData, true);

// Create list with name
createList("category", categoriesData, false);

```